JackieSL1/mini-survey-monkey
# fix: create separate intercepters for /create and /summary endpoints

# Description

**NOTE: This should only be merged AFTER https://github.com/JackieSL1/mini-survey-monkey/pull/70 is merged**

The SurveyStateInterceptor wasn't filtering out requests. I believe this was because Spring was injecting the same dependency twice, making both use the same filtering rules.

This adds two separate interceptor classes to avoid this issue.

## Issues

closes #1

# How to test

Manuals test:
1. Create a Survey
2. Try to go to "localhost:8080/summary/1". You should get an error
3. Open the survey
4. Now, try to go back to "localhost:8080/create/1". You should get an error

Integration tests:
* Run `mvn test`


# Checklist
**I've added tests**

**I've updated the DB schema and UML diagram**
